### DISCUSSION TITLE
SSG Improvements, while using the dev server

### DISCUSSION THREAD

**Discussion by mikestopcontinues** [vercel/next.js#10053] [Ideas]
After experimenting with the new static gen api discussed in #9524, I'd like to suggest a better approach to developing using the dev server.

The problem is, at present, navigating between `getStaticProps` routes causes the dev server to request the results json over and over. This is contrary to production behavior, and it also slows down the development process with extra delays.

@Timer [argued](https://github.com/zeit/next.js/issues/9524#issuecomment-573376506) that during development, it's in the developer's interest to see the most up-to-date data, and thus, the server request is useful. But I don't see a use-case where this would come into play.

1. If you're adjusting code, the dev server is going to hot reload anyway, negating the need to navigate around the app to re-request the json response. If you're clicking around your app, it's likely because your testing behavior, which is slowed by re-requests.
2. And if you're adjusting content that isn't tracked by the dev server (e.g. markdown or some headless cms content), then you're going to have to refresh the browser anyway to see the results of the updates. In this case too, re-requesting the json response doesn't help.

Are there use-cases for re-requests I'm not considering?

Either way, I think we should have the discussion on two fronts: 

1. What should be the default behavior?
2. Should there be a config option to choose between modes?

Thanks. :)

**Comment by Janpot** [comment]
> The problem is, at present, navigating between `getStaticProps` routes causes the dev server to request the results json over and over. This is contrary to production behavior, and it also slows down the development process with extra delays.

So do I understand correctly that the main problem is that `getStaticProps` is doing potentially long running data fetching? Would this be less of a problem if those requests used a cache and follow http cache headers?

**Comment by mikestopcontinues** [comment]
@Janpot That would be a pretty good solution. What do you see as the benefit vs. using the local copy until refresh? Doesn't the hot reload wipe local data already?
